components/camel-sql/src/main/java/org/apache/camel/component/sql/SqlProducer.java

```diff
@@ -171,12 +171,12 @@ public Object doInPreparedStatement(PreparedStatement ps) throws SQLException {
                                 total += count;
                             }
                             exchange.getIn().setHeader(SqlConstants.SQL_UPDATE_COUNT, total);
-                            if (manualCommit) {
+                            if (manualCommit && !restoreAutoCommit) {
```

Comment:
I wonder if there is any better way to check if we are running in a JTA transaction?

Reply:
IIRC
>Commit or rollback should only be called if auto commit has not been set to false before.

So it should check with `if (manualCommit && restoreAutoCommit)` ?

Reply:
Ups, seems your right. I'll change it.

Reply:
> I wonder if there is any better way to check if we are running in a JTA transaction?

You're right that there are better ways to identify JTA, but it's only one option why a connection has been set to auto commit before calling sql producer.

Reply:
@davsclaus can we use `exchange.IsTransacted()` ?

Reply:
I think it should be `true` if `sql` is using in a `transacted()` like
```java
from("direct:foo")
    .transacted()
    .to("sql:INSERT INTO ...");
```

Reply:
Yes if using JTA/XA then you need to use a TX manager and setup this via spring or camel-jta, and this requires enabling the transacted DSL in the routes, so we then know `exchange.isTransacted`. 

So if this is the case then this component can do something different.

Reply:
Well, the question is if only JTA is a valid case that autocommit has been set to false before SqlProducer is called. If yes exchange.isTransacted fits better.